elfloader-tool/include/arch-arm/elfloader.h

```diff
@@ -1,17 +1,19 @@
 /*
  * Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
+ * Copyright 2021, HENSOLDT Cyber
  *
  * SPDX-License-Identifier: GPL-2.0-only
  */
 #pragma once
 
 #include <elfloader_common.h>
 
-typedef void (*init_arm_kernel_t)(paddr_t ui_p_reg_start,
-                                  paddr_t ui_p_reg_end,
-                                  uintptr_t pv_offset,
-                                  vaddr_t v_entry,
-                                  paddr_t dtb, uint32_t dtb_size);
+typedef void (*init_arm_kernel_t)(word_t ui_p_reg_start,
+                                  word_t ui_p_reg_end,
+                                  word_t pv_offset,
+                                  word_t v_entry,
+                                  word_t dtb,
+                                  word_t dtb_size);
```

Comment:
I think I'm happy to merge this as is, but just to double-check: why also replace `vaddr_t` and `paddr_t` with `word_t`?  I might be wrong, but I thought they would be set correctly per architecture (as opposed to maybe `uint32_t`), and they carry some information what kind of word it is we're expecting.

Reply:
The rational here is, that this is a low level machine interface. Type information is not really preserved here across binary boundaries. The type are just wisely chosen so this works without side effects. But all we can do here is pass register values (or some stack values that are register-sized). We do not really have a data structure here also. Maybe I should add a comment here that explains this. Or we keep the types here if we pretend we can keep the semantics.

Reply:
Ok. The parameter names are good enough here to indicate what kind of word we're expecting on the other side of the interface.